Bio/Align/__init__.py

```diff
@@ -923,6 +925,405 @@ def sort(self, key=None, reverse=False):
+                qStarts.append(start2)
+                blockSizes.append(count1)
+                for c1, c2 in zip(seq1[start1:end1], seq2[start2:end2]):
+                    if c1 == 'N' or c2 == 'N':
+                        Ns += 1
+                    elif c1 == c2:
+                        match += 1
+                    else:
+                        mismatch += 1
+                start1 = end1
+                start2 = end2
+        blockcount = len(blockSizes)
+        blockSizes = ",".join(map(str, blockSizes)) + ","
+        qStarts = ",".join(map(str, qStarts)) + ","
+        tStarts = ",".join(map(str, tStarts)) + ","
+        words = [str(match),
+                 str(mismatch),
+                 str(repmatch),
+                 str(Ns),
+                 str(Qgapcount),
+                 str(Qgapbases),
+                 str(Tgapcount),
+                 str(Tgapbases),
+                 strand,
+                 Qname,
+                 str(Qsize),
+                 str(Qstart),
+                 str(Qend),
+                 Tname,
+                 str(Tsize),
+                 str(Tstart),
+                 str(Tend),
+                 str(blockcount),
+                 blockSizes,
+                 qStarts,
+                 tStarts,
+                 ]
+        line = "\t".join(words) + "\n"
+        return line
+
+
+class Alignments(object):
+    """Implements an iterator over pairwise alignments returned by the aligner.
+
+    This class also supports indexing, which is fast for increasing indices,
+    but may be slow for random access of a large number of alignments.
+
+    Note that pairwise aligners can return an astronomical number of alignments,
+    even for relatively short sequences, if they align poorly to each other. We
+    therefore recommend to first check the number of alignments, accessible as
+    len(alignments), which can be calculated quickly even if the number of
+    alignments is very large.
+    """
+
+    def __init__(self, seqA, seqB, score, paths):
+        self.seqA = seqA
+        self.seqB = seqB
+        self.score = score
+        self.paths = paths
+        self.index = -1
+
+    def __len__(self):
+        return len(self.paths)
+
+    def __getitem__(self, index):
+        if index == self.index:
+            return self.alignment
+        if index < self.index:
+            self.paths.reset()
+            self.index = -1
+        while self.index < index:
+            try:
+                alignment = self.next()
+            except StopIteration:
+                raise IndexError('index out of range')
+        return alignment
+
+    def __iter__(self):
+        self.paths.reset()
+        self.index = -1
+        return self
+
+    def __next__(self):
+        path = next(self.paths)
+        self.index += 1
+        alignment = PairwiseAlignment(self.seqA, self.seqB, path, self.score)
+        self.alignment = alignment
+        return alignment
+
+    next = __next__  # Python 2
```

Comment:
Do we need to define the ``.next()`` method at all? Python 2.7 supports the built-in function ``next(...)`` which would work on both.

Or can we make this conditional on Python 2, like:
https://github.com/biopython/biopython/blob/biopython-171/Bio/File.py#L156